src/VernaContextProvider.tsx

```diff
@@ -66,31 +86,121 @@ function genDefaultUiSchema(schema: JSONSchema7, uiSchema: UiSchema) {
+}
+
+function getSelectedUiSchema(
+  defaultUiSchema: UiSchema,
+  defaultSelector: string | undefined,
+): UiSchema {
+  if (defaultSelector && defaultUiSchema?.[defaultSelector]) {
+    return defaultUiSchema[defaultSelector];
+  }
+  return defaultUiSchema;
+}
+
+function getDefaultValues(defaultValues: any, defaultSelector: string | undefined) {
+  if (defaultSelector && defaultValues) return defaultValues[defaultSelector];
+  return defaultValues;
+}
+
 function VernaContextProvider({
   defaultSchema,
   defaultUiSchema,
-  defaultWidget,
+  defaultFormValues,
+  defaultWidgets,
   defaultReadOnly,
+  defaultSelector,
   children,
 }: PropsWithChildren<VernaContextProviderProps>) {
-  const [schema, setSchema] = useState<JSONSchema7>(defaultSchema);
-  const [uiSchema, setUiSchema] = useState<UiSchema>(
+  // The full schemas are not used by the lib itself but may be used for further implementation
+  const [fullSchema, setFullSchema] = useState<JSONSchema7>(defaultSchema);
+  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
     genDefaultUiSchema(defaultSchema, defaultUiSchema),
   );
+  const [schema, _setSchema] = useState<JSONSchema7>(
+    getSelectedSchema(defaultSchema, defaultSelector),
+  );
+  const [uiSchema, _setUiSchema] = useState<UiSchema>(
+    getSelectedUiSchema(defaultUiSchema, defaultSelector),
+  );
+  const [defaultValues, setDefaultValues] = useState(
+    getDefaultValues(defaultFormValues, defaultSelector),
+  );
   const [readOnly, setReadOnly] = useState<boolean>(defaultReadOnly);
-  const [widgets, setWidgets] = useState<WidgetType>(defaultWidget);
+  const [widgets, setWidgets] = useState<WidgetsType>(defaultWidgets);
+  const [selector, setSelector] = useState<string | undefined>(defaultSelector);
+
+  // Schema and uiSchema target a part of fullSchema and fullUiSchema selected by the selector
+  function setSchema(newSchema: JSONSchema7) {
+    if (!selector) {
+      setFullSchema(newSchema);
+      return;
+    }
+    const newFullSchema = { ...fullSchema };
+    if (newFullSchema['properties']) {
+      newFullSchema.properties[selector] = newSchema;
+      _setSchema(newSchema);
+    }
+    setFullSchema(newFullSchema);
+  }
+
+  function setUiSchema(newUiSchema: UiSchema) {
+    if (!selector) {
+      setFullUiSchema(newUiSchema);
+      _setUiSchema(newUiSchema);
+      return;
+    }
+    const newFullUiSchema = { ...fullUiSchema };
+    newFullUiSchema[selector] = newUiSchema;
+    _setUiSchema(newUiSchema);
+    setFullUiSchema(newFullUiSchema);
+  }
+
+  useEffect(() => {
+    if (selector) {
+      if (fullSchema['properties']) setSchema(fullSchema['properties'][selector] as JSONSchema7);
+      if (fullUiSchema) setUiSchema(fullUiSchema[selector]);
+      if (defaultFormValues) setDefaultValues(defaultFormValues[selector]);
+    } else {
+      _setSchema(fullSchema);
+      _setUiSchema(fullUiSchema);
+      setDefaultValues(defaultFormValues);
+    }
+  }, [selector]);
```

Comment:
It's weird to set defaultValues here. If you do that, we will not be able to manage form state later without removing this lines.
```suggestion
    if (selector) {
      if (fullSchema['properties']) setSchema(fullSchema['properties'][selector] as JSONSchema7);
      if (fullUiSchema) setUiSchema(fullUiSchema[selector]);
    } else {
      _setSchema(fullSchema);
      _setUiSchema(fullUiSchema);
    }
  }, [selector]);
```